packages/contracts-periphery/contracts/foundry-tests/OptimistInviter.t.sol

```diff
@@ -0,0 +1,691 @@
+            address(carolERC1271Wallet),
+            bytes32("optimist.can-invite"),
+            abi.encode(3)
+        );
+
+        vm.prank(alice_inviteGranter);
+        optimistInviter.setInviteCounts(addresses, 3);
+
+        assertEq(_getInviteCount(bob), 3);
+        assertEq(_getInviteCount(sally), 3);
+        assertEq(_getInviteCount(address(carolERC1271Wallet)), 3);
+    }
+
+    /**
+     * @notice Bob, who is not the invite granter, should not be able to issue invites.
+     */
+    function test_grantInvites_nonAdminAddingInvites_reverts() external {
+        address[] memory addresses = new address[](2);
+        addresses[0] = bob;
+        addresses[1] = sally;
+
+        vm.expectRevert("OptimistInviter: only invite granter can grant invites");
+        vm.prank(bob);
+        optimistInviter.setInviteCounts(addresses, 3);
+    }
+
+    /**
+     * @notice Sally should be able to commit an invite given by by Bob.
+     */
+    function test_commitInvite_committingForYourself_succeeds() external {
+        _grantInvitesTo(bob);
+        (, bytes memory signature) = _issueInviteAs(bobPrivateKey);
+
+        vm.prank(sally);
+        bytes32 hashedSignature = keccak256(abi.encode(sally, signature));
+        optimistInviter.commitInvite(hashedSignature);
+
+        assertTrue(optimistInviter.commitments(hashedSignature));
+    }
+
+    /**
+     * @notice Sally should be able to Bob's for a different claimer, Eve.
+     */
+    function test_commitInvite_committingForSomeoneElse_succeeds() external {
+        _grantInvitesTo(bob);
+        (, bytes memory signature) = _issueInviteAs(bobPrivateKey);
+
+        vm.prank(sally);
+        bytes32 hashedSignature = keccak256(abi.encode(eve, signature));
+        optimistInviter.commitInvite(hashedSignature);
+
+        assertTrue(optimistInviter.commitments(hashedSignature));
+    }
+
+    /**
+     * @notice Bob issues signature, and Sally claims the invite. Bob's invite count should be
+     *         decremented, and Sally should be able to mint.
+     */
+    function test_claimInvite_succeeds() external {
+        _grantInvitesTo(bob);
+        _issueThenClaimShouldSucceed(bobPrivateKey, sally);
+    }
+
+    /**
+     * @notice Bob issues signature, and Ted commits the invite for Sally. Eve claims for Sally.
+     */
+    function test_claimInvite_claimForSomeoneElse_succeeds() external {
+        _grantInvitesTo(bob);
+        (
+            OptimistInviter.ClaimableInvite memory claimableInvite,
+            bytes memory signature
+        ) = _issueInviteAs(bobPrivateKey);
+
+        _commitInviteAs(sally, signature);
+
+        vm.prank(ted);
+        optimistInviter.commitInvite(keccak256(abi.encode(sally, signature)));
+
+        vm.expectEmit(true, true, true, true, address(attestationStation));
+        emit AttestationCreated(
+            address(optimistInviter),
+            sally,
+            bytes32("optimist.can-mint-from-invite"),
```

Comment:
would prefer to make a call to the contract to get the bytes32 topic values rather than hardcoding them because it makes it more portable